### PR TITLE
Fix `lintOnFly` using `tempFile`

### LINKED ISSUE #61: `lintOnFly` does not work in 2.0.0
I've published a package with working `lintOnFly` called `linter-scss-lint-caseywebdev`

The updated code can be seen at https://github.com/caseywebdev/linter-scss-lint

### PR DESCRIPTION
**Pull request by SpainTrain**
* Lint a temporary file with the current buffer contents to fix on-the-fly linting

Closes #61

### CODE REVIEW

File: lib/init.coffee [hunk 1]
```diff
@@ -28,29 +28,30 @@ module.exports =
...
-          lint = try JSON.parse stdout
-          throw new TypeError(stdout) unless lint?
-          return [] unless lint[filePath]
-          return lint[filePath].map (msg) ->
-            line = (msg.line || 1) - 1
-            col = (msg.column || 1) - 1
+        tempFile path.basename(filePath), editor.getText(), (tmpFilePath) =>
+          config = findFile path.dirname(filePath), '.scss-lint.yml'
+          params = [
+            tmpFilePath,
```
Comment (SpainTrain) [inline lib/init.coffee]: Since the diff is not smart, I will point out actual changed lines

:arrow_left:

File: lib/init.coffee [hunk 2]
```diff
@@ -28,29 +28,30 @@ module.exports =
...
+            if config? then "--config=#{config}",
+            @additionalArguments.split(' ')...
+          ].filter((e) -> e)
+          throw new TypeError(
+            "Error linting #{filePath}: No 'scss-lint' executable specified"
+          ) if @executablePath is ''
+          return helpers.exec(@executablePath, params).then (stdout) ->
+            lint = try JSON.parse stdout
+            throw new TypeError(stdout) unless lint?
+            return [] unless lint[tmpFilePath]
```
Comment (SpainTrain) [inline lib/init.coffee]: :arrow_left:

File: lib/init.coffee [hunk 3]
```diff
@@ -28,29 +28,30 @@ module.exports =
...
+            return lint[tmpFilePath].map (msg) ->
```
Comment (SpainTrain) [inline lib/init.coffee]: :arrow_left:

**Comment by steelbrain** [comment]
IMO, it would be more readable if you would just split the lint function or do something similar. Also is there any `textEditor.isSaved` API? If yes, then you should use that and avoid creating a temp file when it's saved.

**Comment by SpainTrain** [comment]
> avoid creating a temp file when it's saved.

I agree, but can we just create an enhancement ticket for this and address it later?  Considering the 1000s of temp files a user will be creating during OTF linting, not writing one on save is a pretty marginal (and human imperceptible) optimization.

> it would be more readable if you would just split the lint function

Agreed, but was keeping this PR scoped to fixing OTF.  Can open a refactor ticket to clean this thing up.  Focusing for now on getting folks functionality they expect (e.g., #73 #61)

LMK if we are :shipit: and I will create said issues in repo tracker.

**Comment by steelbrain** [comment]
The code LGTM. so if it works. :shipit:

**Comment by SpainTrain** [comment]
https://github.com/AtomLinter/linter-scss-lint/issues/76

and

https://github.com/AtomLinter/linter-scss-lint/issues/77

**Comment by SpainTrain** [comment]
Also can you merge?  I do not have collaborator :sweat_smile:

**Comment by steelbrain** [comment]
:tada:

**Pull request merged**